js/react_native/lib/backend.ts

```diff
@@ -0,0 +1,167 @@
+          return Int32Array;
+        case 'bool':
+          return Int8Array;
+        case 'float64':
+          return Float64Array;
+        case 'int64':
+          /* global BigInt64Array */
+          /* eslint no-undef: ["error", { "typeof": true }] */
+          return BigInt64Array;
+        default:
+          throw new Error(`unsupported type: ${type}`);
+      }
+    };
+
+class OnnxruntimeSessionHandler implements SessionHandler {
+  #inferenceSession: Binding.InferenceSession;
+  #key: string;
+
+  inputNames: string[];
+  outputNames: string[];
+
+  constructor(path: string) {
+    this.#inferenceSession = binding;
+    this.#key = path;
+    this.inputNames = [];
+    this.outputNames = [];
+  }
+
+  async loadModel(options: InferenceSession.SessionOptions): Promise<void> {
+    try {
+      // load a model
+      const results: Binding.ModelLoadInfoType = await this.#inferenceSession.loadModel(this.#key, options);
+      // resolve promise if onnxruntime session is successfully created
+      if (results.key !== this.#key) {
+        throw new Error('Session key is invalid');
+      }
+
+      this.inputNames = results.inputNames;
+      this.outputNames = results.outputNames;
+      return Promise.resolve();
+    } catch (e) {
+      throw new Error('Can\'t load a model');
+    }
+  }
+
+  async dispose(): Promise<void> {
+    return Promise.resolve();
+  }
+
+  startProfiling(): void {
+    // TODO: implement profiling
+  }
+  endProfiling(): void {
+    // TODO: implement profiling
+  }
+
+  async run(feeds: SessionHandler.FeedsType, fetches: SessionHandler.FetchesType, options: InferenceSession.RunOptions):
+      Promise<SessionHandler.ReturnType> {
+    // eslint-disable-next-line no-async-promise-executor
+    return new Promise(async (resolve, reject) => {
+      try {
+        // Java API doesn't support preallocated output names and allows only string array as parameter.
+        const outputNames: Binding.FetchesType = [];
+        for (const fetch in fetches) {
+          if (Object.prototype.hasOwnProperty.call(fetches, fetch)) {
+            outputNames.push(fetch);
+          }
+        }
+        const input = this.encodeFeedsType(feeds);
+        const results: Binding.ReturnType = await this.#inferenceSession.run(this.#key, input, outputNames, options);
+        const output = this.decodeReturnType(results);
+        resolve(output);
+      } catch (e) {
+        reject(e);
+      }
+    });
+  }
+
+  encodeFeedsType(feeds: SessionHandler.FeedsType): Binding.FeedsType {
+    const returnValue: {[name: string]: Binding.EncodedTensorType} = {};
+    for (const key in feeds) {
+      if (Object.hasOwnProperty.call(feeds, key)) {
+        let data: string|string[];
+
+        if (Array.isArray(feeds[key].data)) {
+          data = feeds[key].data as string[];
+        } else {
+          // Base64-encode tensor data
+          data = Buffer.from((feeds[key].data as SupportedTypedArray).buffer).toString('base64');
```

Comment:
use override Buffer.from(arraybuffer, offset, length) to specify byteOffset and byteLength as well